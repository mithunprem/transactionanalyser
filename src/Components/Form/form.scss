form {
  width: 400px;
  text-align: left;

  label {
    font-size: 14px;
    margin: 0 0 8px;
  }
}
