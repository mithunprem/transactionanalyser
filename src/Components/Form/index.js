import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Input from '../../Components/Input';
import Button from '../../Components/Button';
import './form.scss';

export default class Form extends Component {
  state = {
    // Hard coded the form data to be shown by default for representational
    // purpose.
    formData: {
      accountId: "ACC334455",
      fromDate: "2018-10-20T00:00",
      toDate: "2018-10-20T19:00"
    }
  }

  /**
   * Change event handler for the form components.
   */
  handleChange = event => {
    const { value, name } = event.target;

    this.setState(prevState => ({
      formData: {
        ...prevState.formData,
        [name]: value
      }
    }))

    this.props.onChange();
  }

  onSubmit = event => {
    event.preventDefault();
    this.props.onSubmit(this.state.formData);
  }

  render() {
    const { accountId, fromDate, toDate } = this.state.formData;

    return (
      <Fragment>
        <form onSubmit={this.onSubmit}>
          <Input
            type="text" placeholder="Account ID"
            name="accountId" value={accountId}
            onChange={this.handleChange} />
          <Input
            type="datetime-local" placeholder="From"
            name="fromDate" value={fromDate}
            onChange={this.handleChange} />
          <Input
            type="datetime-local" placeholder="To"
            name="toDate" value={toDate}
            onChange={this.handleChange} />
          <Button type="submit" label="Calculate" />
        </form>
      </Fragment>
    )
  }
}

Form.propTypes = {
  onSubmit: PropTypes.func,
  onChange: PropTypes.func
}
