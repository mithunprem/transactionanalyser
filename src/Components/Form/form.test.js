import React from 'react';
import { create } from 'react-test-renderer';
import Form from '../../Components/Form';

describe("Form", () => {
  test("should render without throwing an error", () => {
    const form = create(<Form />).toJSON()
    expect(form).toMatchSnapshot()
  });
});
