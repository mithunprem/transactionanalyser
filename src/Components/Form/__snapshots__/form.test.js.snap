// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render without throwing an error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input"
  >
    <label
      htmlFor="accountId"
    >
      Account ID
    </label>
    <br />
    <input
      name="accountId"
      onChange={[Function]}
      placeholder="Account ID"
      type="text"
      value="ACC334455"
    />
  </div>
  <div
    className="input"
  >
    <label
      htmlFor="fromDate"
    >
      From
    </label>
    <br />
    <input
      name="fromDate"
      onChange={[Function]}
      placeholder="From"
      type="datetime-local"
      value="2018-10-20T00:00"
    />
  </div>
  <div
    className="input"
  >
    <label
      htmlFor="toDate"
    >
      To
    </label>
    <br />
    <input
      name="toDate"
      onChange={[Function]}
      placeholder="To"
      type="datetime-local"
      value="2018-10-20T19:00"
    />
  </div>
  <button
    className="mt-3 button"
    onClick={[Function]}
    type="submit"
  >
    Calculate
  </button>
</form>
`;
